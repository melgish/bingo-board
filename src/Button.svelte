<script>
  export let disabled = false
</script>

<button on:click type="button" {disabled}>
  <slot />
</button>

<style>
  button {
    font-weight: bold;
    background-color: #600;
    border-color: #300;
    color: white;
    padding: 0.5rem 1rem;
    border-radius: 0.5rem;
  }
  button:disabled {
    background-color: #66000080;
  }
</style>
