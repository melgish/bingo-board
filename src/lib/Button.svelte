<script lang="ts">
  export let disabled = false
  export let type = "button" as "button"
</script>

<button on:click {type} {disabled}>
  <slot />
</button>

<style>
  button {
    font-weight: bold;
    background-color: #600;
    border-color: #300;
    color: white;
    padding: 0.5rem 1rem;
    border-radius: 0.5rem;
  }
  button:disabled {
    background-color: #66000080;
  }
</style>
