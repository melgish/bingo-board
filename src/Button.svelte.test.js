import Button from "./Button.svelte"
import { render, screen } from "@testing-library/svelte"
import userEvent from "@testing-library/user-event"

describe("Button", () => {
  describe("when clicked", () => {
    it("should raise an event", () => {
      const clicked = jest.fn()
      const { component } = render(Button)
      component.$on("click", clicked)

      userEvent.click(screen.getByRole("button"))

      expect(clicked).toHaveBeenCalled()
    })
  })
})
