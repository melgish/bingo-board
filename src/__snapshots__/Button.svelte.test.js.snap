// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button when rendered should include the name 1`] = `
<button
  class="svelte-qnnizk"
  name="clicky"
  type="button"
/>
`;
